### DISCUSSION TITLE
How to use Crispulator module in interactive session

### DISCUSSION THREAD

**Discussion by braunb** [tlnagy/Crispulator.jl#66] [Q&A] [answered]
I had played with Crispulator in 2020 and am now dusting it back off. First, I am very glad to see it run seamlessly with the current version of Julia. I'm sure that refactoring was not trivial, but it is much appreciated. I can run it from the zsh shell using the example configuration file.

I'd like to have more programmatic control of the simulations, so I want to dive more into the system than simply running it from the shell. I have used the `Pkg.develop` line suggested in the docs to install Crispulator as a module in my base Julia environment (running MacOS). However, I am not sure how to really use it this way. One thing I am coming up against is that, when running from the command line, a lot of work is done in `parse_yaml.jl` and `runconfig.jl`, but these are not exposed as part of the module. I could download the repository and run these scripts (using `include`), but I'm not sure if that is really the right way to go. 

My impression is that I should write the configuration code myself, using `parse_yaml.jl` as a guide. For example, I can start with
`s = FacsScreen()`
And then modify it as desired. But I though I should see if @tlnagy agrees.

Any advice for using Cripsulator in my custom Julia scripts?

**Comment by tlnagy** [comment, marked as answer]
> For example, I can start with 
> `s = FacsScreen()`
> And then modify it as desired. But I thought I should see if @tlnagy agrees.

That's exactly how I envisioned folks using `Crispulator` beyond the basic simulations. I recommend using it in an interactive environment like Jupyter Notebooks (via `IJulia`) or `Pluto.jl` and building a simulation as you describe. A basic example of this approach is [described in the "Custom Simulation" section of the docs.](https://tamasnagy.com/Crispulator.jl/dev/custom/)
